src/components/useColorScheme.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { IGardenTheme } from '@zendeskgarden/react-theming';
+import { ColorScheme } from './useColorSchemeContext';
+
+export const useColorScheme = (initialState?: ColorScheme) => {
```

Comment:
Along with `useColorSchemeContext`, I think this hook could be a nice addition to `@zendeskgarden/react-theming` for external standard re-use.